WooCommerce/Classes/Analytics/WaitingTimeTracker.swift

```diff
@@ -0,0 +1,106 @@
+import Foundation
+import Combine
+
+class WaitingTimeTracker {
+    private let waitingTimeout: TimeInterval
+
+    @Published private(set) var currentState: State = .idle
```

Comment:
Do you expect this variable to be observed from the outside world? If not, I think you could use a `didSet` observation instead of using a combine observable!